app/code/Magento/Cms/Api/BlockRepositoryInterface.php

```diff
@@ -32,6 +32,16 @@ public function save(Data\BlockInterface $block);
     public function getById($blockId);
 
     /**
+     * Load Block data by given Block identifier
+     *
+     * @param string $blockIdentifier
+     * @param int $storeId
+     * @return \Magento\Cms\Api\Data\BlockInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getByIdentifier($blockIdentifier, $storeId);
```

Comment:
Unfortunately, due to backwards compatibility policy, currently we can not add any new methods to the interfaces, specially the ones marked with the `@api` annotation. You may extract this method to a separate interface/class, if it is really needed.